tests/providers/amazon/aws/sensors/test_step_function_execution.py

```diff
@@ -0,0 +1,142 @@
+
+import unittest
+from unittest import mock
+from unittest.mock import MagicMock
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.sensors.step_function_execution import StepFunctionExecutionSensor
+
+TASK_ID = 'step_function_execution_sensor'
+EXECUTION_ARN = 'arn:aws:states:us-east-1:123456789012:execution:'\
+                'pseudo-state-machine:020f5b16-b1a1-4149-946f-92dd32d97934'
+AWS_CONN_ID = 'aws_non_default'
+REGION_NAME = 'us-west-2'
+
+
+class TestStepFunctionExecutionSensor(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_context = MagicMock()
+
+    def test_init(self):
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        self.assertEqual(TASK_ID, sensor.task_id)
+        self.assertEqual(EXECUTION_ARN, sensor.execution_arn)
+        self.assertEqual(AWS_CONN_ID, sensor.aws_conn_id)
+        self.assertEqual(REGION_NAME, sensor.region_name)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_failed(self, mock_hook):
+        hook_response = {
+            'status': 'FAILED'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_timed_out(self, mock_hook):
+        hook_response = {
+            'status': 'TIMED_OUT'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
+
+    @mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
+    def test_aborted(self, mock_hook):
+        hook_response = {
+            'status': 'ABORTED'
+        }
+
+        hook_instance = mock_hook.return_value
+        hook_instance.describe_execution.return_value = hook_response
+
+        sensor = StepFunctionExecutionSensor(
+            task_id=TASK_ID,
+            execution_arn=EXECUTION_ARN,
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME
+        )
+
+        with self.assertRaises(AirflowException):
+            sensor.poke(self.mock_context)
```

Comment:
You can use `parameterized.expand`
```python
@parameterized.expand([('FAILED',),('TIMED_OUT',),('ABORTED',)])
@mock.patch('airflow.providers.amazon.aws.sensors.step_function_execution.StepFunctionHook')
def test_error(self, mock_status, mock_hook):
    ...
```
to test it three times each with a different status.